tests/test_data_samples_order.py

```diff
@@ -62,17 +81,31 @@ def _shuffle(items):
     return res
 
 
-@pytest.mark.xfail(reason='data samples relative order is not preserved (yet)')
-def test_traintuple_data_samples_relative_order(factory, client, default_dataset):
-    data_sample_keys = _shuffle(default_dataset.train_data_sample_keys)
+@pytest.fixture
+def dataset(factory, client):
+    """Creates a pass through dataset only handling folder names, not actual data."""
+    # create dataset
+    spec = factory.create_dataset(py_script=OPENER_SCRIPT)
+    dataset = client.add_dataset(spec)
+
+    # create train data samples
+    for i in range(4):
```

Comment:
If I understand well, this test has ~4% chance of emitting a false positive. I think this is too high, and could be decrease without costing too much. Maybe 8 data samples?

Reply:
What do you think of replacing the _shuffle method by something like:
```python
def _shuffle(items):
    items = sorted(items)
    return [items[1], items[0]] + items[2:]
```

This way this can never be in alphabetical order or any kind of order.

Reply:
Seems like a good option.

Could you add a comment to the shuffle function to explain its goal (nothing more than rewording _This way this can never be in alphabetical order or any kind of order._) ?

Reply:
Definitely!

Reply:
Done